### PR TITLE
[Merged by Bors] - feat: use `TryThis` infrastructure for `polyrith`

### PR DESCRIPTION
**Pull request by digama0**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Polyrith.lean [hunk 1]
```diff
@@ -315,7 +315,7 @@ to replace the call to `polyrith` with the appropriate call to
 This returns `none` if this was a "dry run" attempt that does not actually invoke sage.
```
Comment (eric-wieser) [inline Mathlib/Tactic/Polyrith.lean:315]: This docstring is no longer accurate

Comment (hrmacbeth) [reply Mathlib/Tactic/Polyrith.lean:315]: @eric-wieser I'll try to address your concern to save Mario some time, but what part of the docstring is not longer accurate in your opinion?

Comment (eric-wieser) [reply Mathlib/Tactic/Polyrith.lean:315]: It looks too me like it no longer rerurns `none` as the return type is no longer `Option`.

Comment (semorrison) [reply Mathlib/Tactic/Polyrith.lean:315]:
```suggestion
Returns `.error g` if this was a "dry run" attempt that does not actually invoke sage.
```

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**